### PR TITLE
Refine CUDA profiler and delete the test file.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/Paddle/issues/6714
Fix https://github.com/PaddlePaddle/Paddle/issues/6033

- Now the config file is created in [C++ code](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/platform/cuda_profiler.h#L24), move it to Python code.
- Delete the file in Python generated in the unit testing.

### CODE REVIEW

File: python/paddle/v2/fluid/profiler.py [hunk 1]
```diff
@@ -8,9 +9,20 @@
...
+    "memtransferdir",
+    "memtransferhostmemtype",
     "streamid",
+    "cacheconfigrequested",
+    "cacheconfigrequested",
+    "cacheconfigrequested",
     "enableonstart 0",
     "conckerneltrace",
+    "active_warps",
+    "active_warps",
```
Comment (chengduoZH) [inline python/paddle/v2/fluid/profiler.py]: Why "active_warps" appeared two times, and "cacheconfigrequested"  appeared three times.

File: python/paddle/v2/fluid/profiler.py [hunk 2]
```diff
@@ -30,17 +42,21 @@ def cuda_profiler(output_file, output_mode=None, config=None):
...
     if output_mode is None:
         output_mode = 'csv'
     if output_mode not in ['kvp', 'csv']:
         raise ValueError("The output mode must be 'kvp' or 'csv'.")
     config = NVPROF_CONFIG if config is None else config
-    core.nvprof_init(output_file, output_mode, config)
+    config_file = 'nvprof_config_file'
+    with open(config_file, 'wb') as fp:
+        fp.writelines(["%s\n" % item for item in config])
+    core.nvprof_init(output_file, output_mode, config_file)
```
Comment (chengduoZH) [inline python/paddle/v2/fluid/profiler.py]: Should we check whether `output_file` exists? If `output_file` has already existed, does this profile result to rewrite the original content of the `output_file`?

File: python/paddle/v2/fluid/profiler.py (hunk 1 above)
Comment (qingqing01) [reply python/paddle/v2/fluid/profiler.py]: Fix, Thanks!

File: python/paddle/v2/fluid/profiler.py (hunk 2 above)
Comment (qingqing01) [reply python/paddle/v2/fluid/profiler.py]: I do not change the code, all you mentioned can be done by the users when using this interface. Thanks!

**Comment by chengduoZH** [comment]
Too much options will affect the speed of the program.

**Comment by qingqing01** [comment]
@chengduoZH Thanks for your careful review. I resume the config after a verification.

**Review by chengduoZH — APPROVED** [review]
LGTM+

**Pull request merged**